### ISSUE TITLE
missing required parameters recaptcha!

### ISSUE THREAD

**Issue by devchris79** [mageplaza/magento-2-google-recaptcha#17]
When customers are logging in they are periodically seeing the error "missing required parameters recaptcha!" it appears the error is occuring when a customer attempts a login quickly (using saved credentials). Recaptcha seems to lag behind the rest of the page load and I think the form is submitted before load completion. I don't know if there is anything that can be done with this..

**Comment by bien119** [comment]
hi @devchris79, Please make sure that you are using the true invisible recaptcha key.
I suppose that you are misunderstanding betwwen invisible key and visible key.

**Comment by devchris79** [comment]
Hi @bien119 I have both keys defined, the page works fine when the full page load is waited on. Its when the form is submitted before page load completion.

**Comment by bien119** [comment]
Oh, you have to wait for the recaptcha loaded. If the full page is not completely loaded, not sure that recaptcha  completely loaded.

**Comment by devchris79** [comment]
A complete load of the login page is taking 2.3 seconds on my connection, however it may be that the customer has a slower connection that it becomes more obvious.
It looks like some some of the recaptcha functionality is being done last after the initial page load in around 1 second.

**Comment by bien119** [comment]
If captcha is loaded with 100%, Is everything fine?

**Comment by devchris79** [comment]
Yes everything works fine if you wait for the recaptcha load period, just to be clear I think the extension is great and I don't know even if anything could be changed, but it would just improve the user experience if the recaptcha could be loaded before a form post was possible.

**Comment by bien119** [comment]
@devchris79 I know your ideal to improve customer experience but the captcha is loaded after the form is complete is the process and Obligatory.

**Comment by devchris79** [comment]
@bien119 thanks for answering the question, if that's just the way it is then fair enough.

**Comment by kriswen** [comment]
@bien119  i'm using google reCAPTCHA v3. How do I confirm i'm using "true invisible recaptcha key"? Thank you

**Comment by jasonhildebrand** [comment]
I have installed on my site, running 2.2.11.  Customers are able to log in, but they see the "missing required parameters recaptcha!" message on the first page after logging in.

I used Chrome devtools to check the login process, and see that there is a POST followed by a GET to `/index.php/customer/account/loginPost`.
![login-requests](https://user-images.githubusercontent.com/7389806/101679959-a4527380-3a25-11eb-8302-e9f12c3f23b0.png)

The POST request contains the form fields including g-recaptcha-response.  The GET only has a single query string parameter to set the store view (we have two store views).

I believe the error is resulting from this GET request.

My solution was to modify `module-google-recaptcha/Observer/Captcha.php` and add the following lines to the top of the `execute()` method in (above the existing code):

        if ($_SERVER['REQUEST_METHOD'] != 'POST') {    
            return;    
        }    

I would be happy to prepare a PR is this is an acceptable fix.
